### PR TITLE
fix(Scripts/BlackTemple): Remove Naj'entus Spines from players on engage.

### PR DESCRIPTION
**Pull request by heyitsbench**
## Changes Proposed:

This PR proposes changes to:
-  [X] Scripts (bosses, spell scripts, creature scripts).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/20168
- Closes https://github.com/chromiecraft/chromiecraft/issues/7407

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)

## Tests Performed:

This PR has been:
- [X] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by TheSCREWEDSoftware** [comment]
Assuming that I needed more people for the mechanic, I spawned the item, killed the boss, died and while dead the item was still in my bag but when i got revived it was gone.

My testing: [Youtube](https://www.youtube.com/watch?v=bXC-GtahzBU&ab_channel=TheSCREWEDSoftware)

**Comment by heyitsbench** [comment]
Will need to test on retail, but I suspect maybe when the boss dies the spines should be removed as well. Thank you for the testing. ❤️

**Comment by heyitsbench** [comment]
Tested on retail PTR. Killing Naj'entus did not remove the Spines from my inventory. Killing Supremus did not remove the Spines from my inventory. Dying did not remove the Spines from my inventory, nor did releasing spirit (which on retail repops at the entrance of the instance). Leaving the instance *did* remove the Spines from my inventory.

**Comment by heyitsbench** [comment]
> If you do test in retail, if you do need more than 1 person, give me a mesasge

I appreciate the offer, I'll keep it in mind in the future. ❤️

**Comment by Rorschach91** [comment]
> Tested on retail PTR. Killing Naj'entus did not remove the Spines from my inventory. Killing Supremus did not remove the Spines from my inventory. Dying did not remove the Spines from my inventory, nor did releasing spirit (which on retail repops at the entrance of the instance). Leaving the instance _did_ remove the Spines from my inventory.

I could have make a mistake this time. Anyway if we want to be sure, it could be a good idea to try on Cata PTR.
If I can be honest, this time I have confirmed the issue because I remebered that the spires vanish when you wipe or when the fight is finished.

File: src/server/scripts/Outland/BlackTemple/boss_warlord_najentus.cpp [hunk 1]
```diff
@@ -64,6 +70,10 @@ struct boss_najentus : public BossAI
 
         BossAI::JustEngagedWith(who);
         Talk(SAY_AGGRO);
+        me->GetMap()->DoForAllPlayers([&](Player* player)
+        {
+            player->DestroyItemCount(ITEM_NAJENTUS_SPINE, 5, true);
```
Comment (Nefertumm) [inline src/server/scripts/Outland/BlackTemple/boss_warlord_najentus.cpp:75]: Can you get more than 5?

**Comment by heyitsbench** [comment]
> If I can be honest, this time I have confirmed the issue because I remebered that the spires vanish when you wipe or when the fight is finished.

I hope I don't need to say this, but memory is not a source we use for changes.

File: src/server/scripts/Outland/BlackTemple/boss_warlord_najentus.cpp (hunk 1 above)
Comment (heyitsbench) [reply src/server/scripts/Outland/BlackTemple/boss_warlord_najentus.cpp:75]: Not without mangling the item template, no.

**Review by Nefertumm — APPROVED** [review]

**Comment by Rorschach91** [comment]
> > If I can be honest, this time I have confirmed the issue because I remebered that the spires vanish when you wipe or when the fight is finished.
> 
> I hope I don't need to say this, but memory is not a source we use for changes.

Yeah I know, I had my head in the clouds at that moment. I'm sorry.

**Comment by TheSCREWEDSoftware** [comment]
Can't send the proper time since I'm on the phone I was Watching ptr run of BT it seems you lose the item when you wipe or if he resets. 

https://youtu.be/5Jont3SGoTo?si=VHV_CpBORMVKbnnC

17:23 - he gets the spine 
Between reset happens 
20:38 - his inventory he had spine but didn't use it 

I think it does make sense since she a boss mechanic item to only be present while the fight is going to on. So maybe changing the condition to check if the encountered was interrupted (by killing him, wiping or resetting)

**Comment by Rorschach91** [comment]
Minute 17:29 He had the spine in his inventory......
Inv 1729

......Wipe. He save himself with invisibility.....

minute 20:38 he open the inventory and the spine is disappeared.

Inv 2038

So the fix is valid.

**Comment by Rorschach91** [comment]
I tested anyway. Tecnically the result is the same, but if I want to be accurate, the spine should be removed from the inventory when the bossfight is resetted.
The script works anyway.

**Comment by Nyeriah** [comment]
Are we sure no spells are involved with this? I wanted to check spell dbc for spells using the spine as reagent (like Kael)

**Comment by heyitsbench** [comment]
> Are we sure no spells are involved with this?

Closest thing I can spot is [Remove Naj'entus Spines](https://wowgaming.altervista.org/aowow/?spell=40354) (40354) which is a serverside spell with no listed effects. I can probably make use of it for this mechanic if desired.

**Comment by heyitsbench** [comment]
> [YouTube PTR run]

Thank you for providing the recording, I'll try to see what I can do.

**Comment by Nyeriah** [comment]
> > Are we sure no spells are involved with this?
> 
> Closest thing I can spot is [Remove Naj'entus Spines](https://wowgaming.altervista.org/aowow/?spell=40354) (40354) which is a serverside spell with no listed effects. I can probably make use of it for this mechanic if desired.

Probably, server side spells can’t be sniffed so it’s possible that it’s used to remove them (specially considering the name).

**Comment by TheSCREWEDSoftware** [comment]
Does anyone here have a recent build of TC or other emulator to see if the behaviour there matches more to the retail?

**Comment by heyitsbench** [comment]
For now, I've put another spine removal with the boss's reset, though without testing (probably on Cata Classic) or more recordings, I'm not sure if it's the correct spot for it to go. Also made use of the serverside spell.

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
What about death though? When the boss dies?

**Comment by heyitsbench** [comment]
> What about death though? When the boss dies?

We can make assumptions about it or someone can hunt down a recording. I know from retail testing that the boss dying does not remove the spines, but I don't know for TBC behavior. I figure it doesn't do any harm and there's not yet any evidence for it, so I didn't include it.

**Comment by Nyeriah** [comment]
Fine by me

**Pull request merged**